bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/CustomButtonDialog.java

```diff
@@ -92,22 +98,46 @@ protected Control createDialogArea(Composite parent) {
 		}
 
 		this.customMessageInputField.setLayoutData(customMessageInputFieldData);
-		if (this.customMistake != null) { // Don't display the spinner if points are determined internally.
+		this.customMessageInputField.setText(Objects.requireNonNullElse(this.customMessage, ""));
+		if (this.customMistake != null || this.forcePenaltyField) { // Don't display the spinner if points are
+																	// determined internally.
 			final Label customPenaltyLabel = new Label(comp, SWT.RIGHT);
 			customPenaltyLabel.setText("Custom Penalty: ");
 			this.customPenaltyInputField = new Spinner(comp, SWT.SINGLE | SWT.BORDER);
 			this.customPenaltyInputField.setDigits(1);
 			this.customPenaltyInputField.setIncrement(5);
 			this.customPenaltyInputField.setLayoutData(data);
+
+			// Multiply by 10 because the selection internally stores the selection as an
+			// integer ignoring the decimal point. (0.5 ==> 5, 1 ==> 10)
+			this.customPenaltyInputField.setSelection((int) Objects.requireNonNullElse(customPenalty * 10, 0d).doubleValue());
```

Comment:
This leads to a NullpointerException when clicking on "Custom Penalty"
```suggestion
			this.customPenaltyInputField.setSelection((int) Objects.requireNonNullElse(customPenalty, 0d).doubleValue() * 10);
```

Please check whether suggestion might work :)

Reply:
Works :)